core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java

```diff
@@ -1601,6 +1602,8 @@
   // As Short data type is used for storing the length of a column during data processing hence
   // the maximum characters that can be supported should be less than Short max value
   public static final int MAX_CHARS_PER_COLUMN_DEFAULT = 32000;
+  // todo: use infinity first, will switch later
+  public static final int MAX_CHARS_PER_COLUMN_INFINITY = -1;
```

Comment:
why this is -1?? as per previous discussion we can support max 67104 chars

Reply:
As I mentioned in another PR, better not to introduce this limits.
-1 means that the parser can parse infinity characters.